module/polls/helper.js

```diff
@@ -0,0 +1,745 @@
+                (
+                    {
+                      _id: pollId
+                    },
+                    [    
+                        "questions"
+                    ]
+                )
+
+                if (!pollQuestions.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+               
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLL_QUESTIONS_FETCHED,
+                    data: pollQuestions[0].questions
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+     /**
+     * Get the poll questions by link.
+     * @method
+     * @name getPollQuestionsByLink
+     * @param {String} link - link 
+     * @param {String} userId - userId
+     * @returns {JSON} - poll questions and options.
+     */
+
+    static getPollQuestionsByLink(link= "", userId= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (link == "") {
+                    throw new Error(messageConstants.apiResponses.LINK_REQUIRED_CHECK)
+                }
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                let pollQuestions = await this.pollDocuments
+                (
+                    {
+                      link: link
+                    },
+                    [    
+                        "questions",
+                        "status",
+                        "endDate"
+                    ]
+                )
+                
+                if (!pollQuestions.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                if (new Date() > new Date(pollQuestions[0].endDate)) {
+                    
+                    if (pollQuestions[0].status == messageConstants.common.ACTIVE_STATUS) {
+                        await this.updatePollDocument
+                        (
+                            { link : link },
+                            { $set : { status: messageConstants.common.INACTIVE_STATUS } }
+                        )
+                    }
+                    
+                    throw new Error(messageConstants.apiResponses.LINK_IS_EXPIRED)
+                }
+                else {
+                
+                let result = {
+                    pollId: pollQuestions[0]._id,
+                    questions: pollQuestions[0].questions,
+                    submissionId: ""
+                }
+           
+                let pollSubmissionDocument = await database.models.pollSubmissions.find
```

Comment:
@deepakr123 - Use a helper function of pollSubmissions module.

Reply:
resolved